app/src/main/java/com/eveningoutpost/dexdrip/Services/G5CollectionService.java

```diff
@@ -563,8 +553,8 @@ private void processNewTransmitterData(TransmitterData transmitterData, long tim
         Sensor.updateBatteryLevel(sensor, transmitterData.sensor_battery_level);
         android.util.Log.i("timestamp create", Long.toString(transmitterData.timestamp));
 
-        BgReading.create(transmitterData.raw_data, transmitterData.filtered_data, this, transmitterData.timestamp);
-        transmitterData.save();
+        BgReading.create(transmitterData.raw_data, filtered_data, this, transmitterData.timestamp);
+
 
     }
 
```

Comment:
Ah, just saw your note Adrian - is deduping occurring elsewhere? I haven't seen it happening, which is why I implemented it that way. I would hold off on these changes for now, revisit once we get more stable process and startup with G5 users.

----
John Costik :: Lead Software Developer
Center for Clinical Innovation :: University of Rochester
(585) 350-5391<tel:(585)%20350-5391>




On Thu, Mar 31, 2016 at 3:21 PM -0700, "John Costik" <notifications@github.com<mailto:notifications@github.com>> wrote:


In app/src/main/java/com/eveningoutpost/dexdrip/Services/G5CollectionService.java<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_StephenBlackWasAlreadyTaken_xDrip-2DExperimental_pull_316-23discussion-5Fr58137513&d=BQMCaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=-S4tVkPALJfUIaQ0q618B_o0Ok9zeQjeO8H8WZ2UpSA&m=bEC20KGYzyKaAP4uY6483hLfBtsaB8g1JYSEg_aQBTg&s=MEaTKPdDEs7BLXbNx5fDmQAlYI8_5X0drXWWS-skICQ&e=>:

> @@ -563,8 +553,8 @@ private void processNewTransmitterData(TransmitterData transmitterData, long tim
>          Sensor.updateBatteryLevel(sensor, transmitterData.sensor_battery_level);
>          android.util.Log.i("timestamp create", Long.toString(transmitterData.timestamp));
>
> -        BgReading.create(transmitterData.raw_data, transmitterData.filtered_data, this, transmitterData.timestamp);
> -        transmitterData.save();
> +        BgReading.create(transmitterData.raw_data, filtered_data, this, transmitterData.timestamp);
> +
>
>      }
>


The last read, time since logic should probably stay put. When the G5 wakes up, the app may acquire raw data multiple times before it goes back to sleep. I put those checks in place because BgReading create was inserting multiple duplicates.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_StephenBlackWasAlreadyTaken_xDrip-2DExperimental_pull_316_files_4d03f34e5cdfaf692057d0c63b0d0318963c58c1-23r58137513&d=BQMCaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=-S4tVkPALJfUIaQ0q618B_o0Ok9zeQjeO8H8WZ2UpSA&m=bEC20KGYzyKaAP4uY6483hLfBtsaB8g1JYSEg_aQBTg&s=O0eacF6LaRmliGW9zyJTP__rzuGfPKX2jSdJxbM6gm4&e=>